File: 90-subsets-ii/90-subsets-ii.py
class Solution:
    def __init__(self):
        self.ans = []
        
    def solve(self, nums, curr, start):
        self.ans.append(curr.copy())

        for i in range(start, len(nums)):
            if ((i == start) or (nums[i] != nums[i - 1])):
                curr.append(nums[i])
                self.solve(nums, curr, i + 1)
                curr.pop()
                
        
    def subsetsWithDup(self, nums: List[int]) -> List[List[int]]:
        nums.sort()
        self.solve(nums, [], 0)
        return self.ans

File: 90-subsets-ii/90-subsets-ii.cpp
class Solution {
public:
    
    void  ss(vector<int>& nums, vector<int> v, vector<vector<int>> &vv, int i)
    {
        vv.push_back(v);
        
        for (int j = i; j < nums.size(); j++) 
        {
            if (j == i || nums[j] != nums[j - 1])
            {
                v.push_back(nums[j]);
                ss(nums, v, vv, j + 1);
                v.pop_back();   
            }
		}
    }
    
    vector<vector<int>> subsetsWithDup(vector<int>& nums) 
    {   
        sort(nums.begin(), nums.end());
        vector<vector<int>> vv;
        vector<int> v;
        ss(nums, v, vv, 0);
        return vv;
    }
};


